src/Matcher/ArrayMatcher.php

```diff
@@ -203,6 +203,7 @@ private function getPropertyAccessor()
         }
 
         $accessorBuilder = PropertyAccess::createPropertyAccessorBuilder();
+        $accessorBuilder->enableExceptionOnInvalidIndex();
```

Comment:
if ``exceptionOnInvalidIndex`` is disabled then ``isReadable`` method returns ``true`` for unexisting properties of array, but if this option is enabled ``isReadable`` method catches all exceptions (eg InvalidIndex) and returns ``false``. That's why I enabled exceptions.
 Unfortunately this method was added in ``2.5``

e.g.
https://github.com/symfony/property-access/blob/2.5/PropertyAccessor.php#L134